source/vir/src/ast.rs

```diff
@@ -499,7 +501,7 @@ pub enum StmtX {
 /// Function parameter
 pub type Param = Arc<Spanned<ParamX>>;
 pub type Params = Arc<Vec<Param>>;
-#[derive(Debug, Clone)]
+#[derive(Debug, ToNode, Clone)]
```

Comment:
Do we also want to apply compact printing to function arguments and the return type?

Reply:
Can you elaborate? The `ToNodeOpts` are propagated in all recursive calls, but maybe that's not what you mean.

Reply:
Using `compact` mode, the following code snippet generates the VIR log below. It seems like `ParamX`'s type information is logged, and I wonder if we want to remove these type information of function arguments and the return type.  If we choose to do that, I am guessing that we might add a handwritten `impl ToNode for ParamX`?

```
spec fn mul(a: u64, b: u64)  -> u64 {
    builtin::mul(a, b)
}
```

```
;; rust_verify/example/assorted_demo.rs:24:6: 24:36 (#23375)
(Function
 (Fun :path "mul." :trait_path None) :mode Spec :typ_bounds () :params (
  (Param :name "a~2" :typ (Typ Boxed (Typ Int (IntRange U 64))) :mode Spec :is_mut false)
  (Param :name "b~4" :typ (Typ Boxed (Typ Int (IntRange U 64))) :mode Spec :is_mut false)
 ) :ret (Param :name "%return" :typ (Typ Int (IntRange U 64)) :mode Spec :is_mut false) :require () :ensure () :body (> Block
  () (> Binary (BinaryOp Arith (ArithOp Mul) 24) (> Var "a~2") (> Var "b~4"))))
```

Reply:
Thanks! I'd argue that type signatures for functions are still fairly critical info, so I'd print it in compact mode too (where I'd interpret `no_type` as "no types of expressions").